exporter/logicmonitorexporter/logs_exporter.go

```diff
@@ -0,0 +1,130 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logicmonitorexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/logicmonitorexporter"
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/logicmonitor/lm-data-sdk-go/api/logs"
+	"github.com/logicmonitor/lm-data-sdk-go/utils"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+type logExporter struct {
+	config          *Config
+	logger          *zap.Logger
+	logIngestClient *logs.LMLogIngest
+	settings        component.TelemetrySettings
+}
+
+const (
+	hostname         = "hostname"
+	hostnameProperty = "system.hostname"
+)
+
+// Create new logicmonitor logs exporter
+func newLogsExporter(cfg component.Config, set component.ExporterCreateSettings) *logExporter {
+	oCfg := cfg.(*Config)
+
+	return &logExporter{
+		config:   oCfg,
+		logger:   set.Logger,
+		settings: set.TelemetrySettings,
+	}
+}
+
+func (e *logExporter) start(_ context.Context, host component.Host) error {
+	client, err := e.config.HTTPClientSettings.ToClient(host, e.settings)
+	if err != nil {
+		return fmt.Errorf("failed to create http client: %w", err)
+	}
+	authParams := utils.AuthParams{
+		AccessID:    e.config.APIToken.AccessID,
+		AccessKey:   e.config.APIToken.AccessKey,
+		BearerToken: e.config.Headers["Authorization"],
+	}
+	options := []logs.Option{
+		logs.WithLogBatchingDisabled(),
+		logs.WithAuthentication(authParams),
+		logs.WithHTTPClient(client),
+		logs.WithEndpoint(e.config.Endpoint),
+	}
+
+	lli, err := logs.NewLMLogIngest(context.Background(), options...)
+	if err != nil {
+		return fmt.Errorf("failed to initialize LMLogIngest: %w", err)
+	}
+	e.logIngestClient = lli
+	return nil
+}
+
+func (e *logExporter) PushLogData(ctx context.Context, lg plog.Logs) error {
+	e.logger.Debug("Pushing log data to Logicmonitor")
+	resourceLogs := lg.ResourceLogs()
+	for i := 0; i < resourceLogs.Len(); i++ {
+		resourceLog := resourceLogs.At(i)
+		libraryLogs := resourceLog.ScopeLogs()
+		for j := 0; j < libraryLogs.Len(); j++ {
+			libraryLog := libraryLogs.At(j)
+			logs := libraryLog.LogRecords()
+			for k := 0; k < logs.Len(); k++ {
+				logMetadataMap := make(map[string]string)
+				resourceMapperMap := make(map[string]string)
+				log := logs.At(k)
+				// Copying resource attributes to log attributes
+				resourceLog.Resource().Attributes().Range(func(k string, v pcommon.Value) bool {
```

Comment:
You should be able to use:
```suggestion
				resourceLog.Resource().Attributes().CopyTo(log.Attributes())
```

Reply:
Yes this is much better and cleaner approach. Thanks for suggestion :)